public/app/features/explore/utils/decorators.ts

```diff
@@ -48,6 +49,9 @@ export const decorateWithFrameTypeMetadata = (data: PanelData): ExplorePanelData
       case 'nodeGraph':
         nodeGraphFrames.push(frame);
         break;
+      case 'flamegraph':
```

Comment:
What about guarding with the flag here? That way users will at least get a table, which I think is a bit better (for knowing what is going on) then getting No Data when you query something that wants flamegraph.